src/shims/fs.rs

```diff
@@ -340,73 +340,68 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(create_link(target, linkpath).map(|_| 0))
     }
 
-    fn stat(
+    fn macos_stat(
         &mut self,
         path_op: OpTy<'tcx, Tag>,
         buf_op: OpTy<'tcx, Tag>,
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
         this.check_no_isolation("stat")?;
+        this.check_platform("macos", "stat")?;
         // `stat` always follows symlinks.
-        this.stat_or_lstat(true, path_op, buf_op)
+        this.macos_stat_or_lstat(true, path_op, buf_op)
     }
 
     // `lstat` is used to get symlink metadata.
-    fn lstat(
+    fn macos_lstat(
         &mut self,
         path_op: OpTy<'tcx, Tag>,
         buf_op: OpTy<'tcx, Tag>,
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
         this.check_no_isolation("lstat")?;
-        this.stat_or_lstat(false, path_op, buf_op)
+        this.check_platform("macos", "lstat")?;
+        this.macos_stat_or_lstat(false, path_op, buf_op)
     }
 
-    fn fstat(
+    fn macos_fstat(
         &mut self,
         fd_op: OpTy<'tcx, Tag>,
         buf_op: OpTy<'tcx, Tag>,
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
         this.check_no_isolation("fstat")?;
-
-        if this.tcx.sess.target.target.target_os.to_lowercase() != "macos" {
-            throw_unsup_format!("The `fstat` shim is only available for `macos` targets.")
-        }
+        this.check_platform("macos", "fstat")?;
 
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
         let metadata = match FileMetadata::from_fd(this, fd)? {
             Some(metadata) => metadata,
             None => return Ok(-1),
         };
-        stat_macos_write_buf(this, metadata, buf_op)
+        macos_stat_write_buf(this, metadata, buf_op)
     }
 
-    fn stat_or_lstat(
+    fn macos_stat_or_lstat(
```

Comment:
This one has no platform check though?

Reply:
well both `stat` and `lstat` do. This function doesn't check for isolation either.

Reply:
Then please add a doc comment saying what checks it expects the caller to have made.